src/palace/manager/sqlalchemy/util.py

```diff
@@ -158,7 +158,10 @@ def numericrange_to_string(r):
     return f"{lower}-{upper}"
 
 
-def numericrange_to_tuple(r):
+NumericRangeTuple = tuple[float | None, float | None]
```

Comment:
This one is a bit tricky. PG has it type hinted as a `float`, because it can be either an `int` or a `float`. In our case right now we only use ranges with `int`. I could update the type hints for `int` and assert it so we fail if we ever do use floats, or just leave it as is.

Reply:
Whatever protection we decide to provide should probably happen in/for `numericrange_to_string`, since that's where the behavior that expects `int`s is.

Feel free to handle it however you think is the most sensible. I just wanted highlight the problem (/me lobs hand grenade and runs away 😂).